src/Features/WCPayPromotion/Init.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Handles wcpay promotion
+ */
+
+namespace Automattic\WooCommerce\Admin\Features\WCPayPromotion;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\WooCommerce\Admin\Loader;
+use Automattic\WooCommerce\Admin\PaymentPlugins;
+
+/**
+ * WC Pay Promotion engine.
+ */
+class Init {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		include_once __DIR__ . '/WCPaymentGatewayPsuedoWCPay.php';
+
+		add_filter( PaymentPlugins::FILTER_NAME, array( __CLASS__, 'possibly_filter_recommended_payment_gateways' ) );
+
+		if ( ! isset( $_GET['page'] ) || 'wc-settings' !== $_GET['page'] || ! isset( $_GET['tab'] ) || 'checkout' !== $_GET['tab'] ) { // phpcs:ignore WordPress.Security.NonceVerification
+			return;
+		}
```

Comment:
Only register the JS and the payment gateway filters when we are rendering the Payment settings page.